dozer-ingestion/src/connectors/mod.rs

```diff
@@ -229,15 +230,18 @@ pub fn get_connector(connection: Connection) -> Result<Box<dyn Connector>, Conne
     }
 }
 
-pub fn get_connector_info_table(connection: &Connection) -> Option<Table> {
+pub fn get_connector_info_table(connection: &Connection) -> Result<Table, ConnectorError> {
     match &connection.config {
-        Some(ConnectionConfig::Postgres(config)) => Some(config.convert_to_table()),
-        Some(ConnectionConfig::Ethereum(config)) => Some(config.convert_to_table()),
-        Some(ConnectionConfig::Snowflake(config)) => Some(config.convert_to_table()),
-        Some(ConnectionConfig::Kafka(config)) => Some(config.convert_to_table()),
-        Some(ConnectionConfig::S3Storage(config)) => Some(config.convert_to_table()),
-        Some(ConnectionConfig::LocalStorage(config)) => Some(config.convert_to_table()),
-        _ => None,
+        Some(ConnectionConfig::Postgres(config)) => match config.replenish() {
+            Ok(conf) => Ok(conf.convert_to_table()),
+            Err(_) => Err(WrongConnectionConfiguration),
```

Comment:
As stated before, we should try to include error context and source in an error variant, so it's actually helpful.

Reply:
sure